src/cloudai/schema/system/slurm/slurm_system.py

```diff
@@ -49,41 +49,35 @@ class SlurmSystem(System):
     """
 
     @classmethod
-    def parse_node_list(cls, node_list: List[str]) -> List[str]:
+    def parse_node_list(cls, node_list: str) -> List[str]:
         """
         Expand a list of node names (with ranges) into a flat list of individual node names, keeping leading zeroes.
 
         Args:
-            node_list (List[str]): A list of node names, possibly including ranges.
+            node_list (str): A list of node names, possibly including ranges.
 
         Returns:
             List[str]: A flat list of expanded node names with preserved
             zeroes.
         """
-        expanded_nodes = []
-        for node in node_list:
-            if "[" in node and "]" in node:
-                prefix, ranges = node.split("[")
-                ranges = ranges.strip("]")
-                range_elements = ranges.split(",")
-                for r in range_elements:
-                    if "-" in r:
-                        start_str, end_str = r.split("-")
-                    else:
-                        # For single nodes, treat the node itself as both start and end.
-                        start_str = end_str = r
-
-                    start, end = int(start_str), int(end_str)
-                    max_length = max(len(start_str), len(end_str))
-
-                    if "-" in r:
-                        expanded_nodes.extend([f"{prefix}{str(i).zfill(max_length)}" for i in range(start, end + 1)])
-                    else:
-                        # For single nodes, append directly with appropriate padding.
-                        expanded_nodes.append(f"{prefix}{start_str.zfill(max_length)}")
-            else:
-                expanded_nodes.append(node)
-        return expanded_nodes
+        nodes = []
+        if node_list:
```

Comment:
To reduce nesting, you can handle this corner case explicitly:
```py
if not node_list:
    return []

...
```

Reply:
sure I updated